pkg/operator/ceph/cluster/predicate_test.go

```diff
@@ -40,3 +42,38 @@ func TestIsHotPlugCM(t *testing.T) {
 	cm.Labels["app"] = "rook-discover"
 	assert.True(t, isHotPlugCM(cm))
 }
+
+func TestCompareNodes(t *testing.T) {
```

Comment:
Please test all cases. I only see one case tested, and it doesn't give me confidence that the comparators are working as we expect them to.

What happens if only the heartbeat time changes?
What happens if only the resource version changes?
What happens if nothing changes?
What happens if a field changes, but not resource version or heartbeat?
What happens if a field changes, and only resource version?
What happens if a field changes, and only heartbeat?